docs/quick-start.md

```diff
@@ -0,0 +1,324 @@
+    switch (job_status) {
+        case JobStatus::Succeeded: {
+            auto result = job_status.get_result();
+            int expected = x + y;
+            if (expected == result) {
+                return 0;
+            } else {
+                std::cerr << "`sum` returned unexpected result. Expected: " << expected
+                        << ". Actual: " << result << '\n';
+                return 1;
+            }
+        }
+        case JobStatus::Failed: {
+            std::pair<std::string, std::string> error_and_fn_name = job.get_error();
+            std::cerr << "Job failed in function " << error_and_fn_name.second << " - "
+                   << error_and_fn_name.first << '\n';
+            return 1;
+        }
+        default:
+            std::cerr << "Job is in unexpected state - " << job_status << '\n';
+            return 1;
+    }
+}
+```
+
+When we submit a task to Spider, Spider returns a `Job`, which represents a scheduled, running, or
+completed task (or `TaskGraph`) in a Spider cluster.
+
+> [!NOTE]
+> `Job`s and `TaskGraph`s will be explained in another guide.
+
+# Building the client
+
+The client can be compiled like any normal C++ application, except that we need to link it to the
+Spider client library. To do so, add the following to `CMakeLists.txt`:
+
+```cmake
+# Add the client
+add_executable(client src/client.cpp)
+
+# Link the spider library to the client
+target_link_libraries(client PRIVATE spider::spider)
+```
+
+To build the client executable, run:
+
+```shell
+cmake -S . -B build
+cmake --build build --parallel $(nproc) --target client
+```
+
+# Setting up a Spider cluster
+
+Before we can run the client, we need to start a Spider cluster. The simplest Spider cluster
+consists of:
+
+* a storage backend;
+* a scheduler instance;
+* and a worker instance.
+
+## Setting up a storage backend
+
+Spider currently supports using MySQL or MariaDB as a storage backend. In this guide, we'll start
+MariaDB in a Docker container:
+
+```shell
+docker run \
+        --detach \
+        --rm \
+        --name spider-storage \
+        --env MARIADB_USER=spider \
+        --env MARIADB_PASSWORD=password \
+        --env MARIADB_DATABASE=spider-storage \
+        --env MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=true \
+        --publish 3306:3306 mariadb:latest
+```
+
+> [!WARNING]
+> When the container above is stopped, the database will be deleted. In production, you should set
+> up a database instance with some form of data persistence.
+
+> [!WARNING]
+> The container above is using hardcoded default credentials that shouldn't be used in production.
+
+Alternatively, if you have an existing MySQL/MariaDB instance, you can use that as well. Simply
+create a database and authorize a user to access it.
+
+## Setting up the scheduler
+
+To build the scheduler, run:
+
+```shell
+cmake -S spider -B spider/build
+cmake --build spider/build --parallel $(nproc) --target spider_scheduler
+```
```

Comment:
Why build spider inside a separate build directory again? The root level `CMakeLists.txt` includes `add_directory(spider)`, which can be used to spider.